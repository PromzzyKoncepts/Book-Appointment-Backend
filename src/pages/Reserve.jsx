import React, { useState } from 'react';

const ReservePage = ({ username, selectedItem }) => {
  const [date, setDate] = useState('');
  const [city, setCity] = useState('');

  const handleSubmit = (event) => {
    event.preventDefault();
    // Here you can send the reservation data to your backend or do any other necessary logic
    console.log(`Reservation for ${selectedItem} on ${date} in ${city} made by ${username}`);
  };

  return (
    <div>
      <h1>Reserve {selectedItem}</h1>
      <form onSubmit={handleSubmit}>
        <label>
          Date:
          <input type="date" value={date} onChange={(event) => setDate(event.target.value)} />
        </label>
        <br />
        <label>
          City:
          <input type="text" value={city} onChange={(event) => setCity(event.target.value)} />
        </label>
        <br />
        <button type="submit">Reserve</button>
      </form>
    </div>
  );
};

export default ReservePage;