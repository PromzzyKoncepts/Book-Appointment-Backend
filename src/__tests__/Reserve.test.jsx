import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/';
import Reserve from '../pages/Reserve';

describe('Reserve Page', () => {
  test('Should render the proper Reserve Page', () => {
    const reservePage = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Reserve />
          </Router>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(reservePage).toMatchSnapshot();
  });
});
