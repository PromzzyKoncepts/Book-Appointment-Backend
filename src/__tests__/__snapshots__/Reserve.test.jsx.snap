// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserve Page Should render the proper Reserve Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gueYoa hjwgPs"
      >
        <form
          class="sc-beqWaB eTVxgX"
        >
          <h1
            class="sc-gLDzan hoMnOv"
          >
            Reserve
          </h1>
          <input
            aria-label="name"
            class="sc-dmqHEX jEDZgU"
            placeholder="user"
            readonly=""
            type="text"
            value=""
          />
          <select
            aria-label="Select label"
            class="select-car"
          />
          <label
            class="sc-hLseeU lkbGtg"
          >
            Enter start-up date
          </label>
          <div
            class="react-date-picker react-date-picker--closed react-date-picker--enabled select-car"
          >
            <div
              class="react-date-picker__wrapper"
            >
              <div
                class="react-date-picker__inputGroup"
              >
                <input
                  hidden=""
                  max="275760-09-13"
                  min="0001-01-01"
                  name="date"
                  style="visibility: hidden; position: absolute; z-index: -999;"
                  type="date"
                  value="2023-04-12"
                />
                <input
                  autocomplete="off"
                  class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
                  data-input="true"
                  inputmode="numeric"
                  max="12"
                  min="1"
                  name="month"
                  placeholder="--"
                  type="number"
                  value="4"
                />
                <span
                  class="react-date-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autocomplete="off"
                  class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
                  data-input="true"
                  inputmode="numeric"
                  max="30"
                  min="1"
                  name="day"
                  placeholder="--"
                  type="number"
                  value="12"
                />
                <span
                  class="react-date-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autocomplete="off"
                  class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
                  data-input="true"
                  inputmode="numeric"
                  max="275760"
                  min="1"
                  name="year"
                  placeholder="----"
                  step="1"
                  type="number"
                  value="2023"
                />
              </div>
              <button
                class="react-date-picker__clear-button react-date-picker__button"
                type="button"
              >
                <svg
                  class="react-date-picker__clear-button__icon react-date-picker__button__icon"
                  height="19"
                  stroke="black"
                  stroke-width="2"
                  viewBox="0 0 19 19"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                class="react-date-picker__calendar-button react-date-picker__button"
                type="button"
              >
                <svg
                  class="react-date-picker__calendar-button__icon react-date-picker__button__icon"
                  height="19"
                  stroke="black"
                  stroke-width="2"
                  viewBox="0 0 19 19"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
          <label
            class="sc-hLseeU lkbGtg"
          >
            Enter Return date
          </label>
          <div
            class="react-date-picker react-date-picker--closed react-date-picker--enabled select-car"
          >
            <div
              class="react-date-picker__wrapper"
            >
              <div
                class="react-date-picker__inputGroup"
              >
                <input
                  hidden=""
                  max="275760-09-13"
                  min="0001-01-01"
                  name="date"
                  style="visibility: hidden; position: absolute; z-index: -999;"
                  type="date"
                  value="2023-04-12"
                />
                <input
                  autocomplete="off"
                  class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
                  data-input="true"
                  inputmode="numeric"
                  max="12"
                  min="1"
                  name="month"
                  placeholder="--"
                  type="number"
                  value="4"
                />
                <span
                  class="react-date-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autocomplete="off"
                  class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
                  data-input="true"
                  inputmode="numeric"
                  max="30"
                  min="1"
                  name="day"
                  placeholder="--"
                  type="number"
                  value="12"
                />
                <span
                  class="react-date-picker__inputGroup__divider"
                >
                  /
                </span>
                <input
                  autocomplete="off"
                  class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
                  data-input="true"
                  inputmode="numeric"
                  max="275760"
                  min="1"
                  name="year"
                  placeholder="----"
                  step="1"
                  type="number"
                  value="2023"
                />
              </div>
              <button
                class="react-date-picker__clear-button react-date-picker__button"
                type="button"
              >
                <svg
                  class="react-date-picker__clear-button__icon react-date-picker__button__icon"
                  height="19"
                  stroke="black"
                  stroke-width="2"
                  viewBox="0 0 19 19"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                class="react-date-picker__calendar-button react-date-picker__button"
                type="button"
              >
                <svg
                  class="react-date-picker__calendar-button__icon react-date-picker__button__icon"
                  height="19"
                  stroke="black"
                  stroke-width="2"
                  viewBox="0 0 19 19"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
          <label
            class="sc-hLseeU lkbGtg"
          >
            Enter your city
          </label>
          <input
            aria-label="city"
            class="sc-dmqHEX jEDZgU"
            name="city"
            required=""
            type="text"
            value=""
          />
          <button
            class="sc-jTrPJq GpDmH"
            type="submit"
          >
            Reserve Now
          </button>
        </form>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gueYoa hjwgPs"
    >
      <form
        class="sc-beqWaB eTVxgX"
      >
        <h1
          class="sc-gLDzan hoMnOv"
        >
          Reserve
        </h1>
        <input
          aria-label="name"
          class="sc-dmqHEX jEDZgU"
          placeholder="user"
          readonly=""
          type="text"
          value=""
        />
        <select
          aria-label="Select label"
          class="select-car"
        />
        <label
          class="sc-hLseeU lkbGtg"
        >
          Enter start-up date
        </label>
        <div
          class="react-date-picker react-date-picker--closed react-date-picker--enabled select-car"
        >
          <div
            class="react-date-picker__wrapper"
          >
            <div
              class="react-date-picker__inputGroup"
            >
              <input
                hidden=""
                max="275760-09-13"
                min="0001-01-01"
                name="date"
                style="visibility: hidden; position: absolute; z-index: -999;"
                type="date"
                value="2023-04-12"
              />
              <input
                autocomplete="off"
                class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
                data-input="true"
                inputmode="numeric"
                max="12"
                min="1"
                name="month"
                placeholder="--"
                type="number"
                value="4"
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                autocomplete="off"
                class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
                data-input="true"
                inputmode="numeric"
                max="30"
                min="1"
                name="day"
                placeholder="--"
                type="number"
                value="12"
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                autocomplete="off"
                class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
                data-input="true"
                inputmode="numeric"
                max="275760"
                min="1"
                name="year"
                placeholder="----"
                step="1"
                type="number"
                value="2023"
              />
            </div>
            <button
              class="react-date-picker__clear-button react-date-picker__button"
              type="button"
            >
              <svg
                class="react-date-picker__clear-button__icon react-date-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="4"
                  x2="15"
                  y1="4"
                  y2="15"
                />
                <line
                  x1="15"
                  x2="4"
                  y1="4"
                  y2="15"
                />
              </svg>
            </button>
            <button
              class="react-date-picker__calendar-button react-date-picker__button"
              type="button"
            >
              <svg
                class="react-date-picker__calendar-button__icon react-date-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="15"
                  width="15"
                  x="2"
                  y="2"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="0"
                  y2="4"
                />
                <line
                  x1="13"
                  x2="13"
                  y1="0"
                  y2="4"
                />
              </svg>
            </button>
          </div>
        </div>
        <label
          class="sc-hLseeU lkbGtg"
        >
          Enter Return date
        </label>
        <div
          class="react-date-picker react-date-picker--closed react-date-picker--enabled select-car"
        >
          <div
            class="react-date-picker__wrapper"
          >
            <div
              class="react-date-picker__inputGroup"
            >
              <input
                hidden=""
                max="275760-09-13"
                min="0001-01-01"
                name="date"
                style="visibility: hidden; position: absolute; z-index: -999;"
                type="date"
                value="2023-04-12"
              />
              <input
                autocomplete="off"
                class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
                data-input="true"
                inputmode="numeric"
                max="12"
                min="1"
                name="month"
                placeholder="--"
                type="number"
                value="4"
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                autocomplete="off"
                class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
                data-input="true"
                inputmode="numeric"
                max="30"
                min="1"
                name="day"
                placeholder="--"
                type="number"
                value="12"
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                autocomplete="off"
                class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
                data-input="true"
                inputmode="numeric"
                max="275760"
                min="1"
                name="year"
                placeholder="----"
                step="1"
                type="number"
                value="2023"
              />
            </div>
            <button
              class="react-date-picker__clear-button react-date-picker__button"
              type="button"
            >
              <svg
                class="react-date-picker__clear-button__icon react-date-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="4"
                  x2="15"
                  y1="4"
                  y2="15"
                />
                <line
                  x1="15"
                  x2="4"
                  y1="4"
                  y2="15"
                />
              </svg>
            </button>
            <button
              class="react-date-picker__calendar-button react-date-picker__button"
              type="button"
            >
              <svg
                class="react-date-picker__calendar-button__icon react-date-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="15"
                  width="15"
                  x="2"
                  y="2"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="0"
                  y2="4"
                />
                <line
                  x1="13"
                  x2="13"
                  y1="0"
                  y2="4"
                />
              </svg>
            </button>
          </div>
        </div>
        <label
          class="sc-hLseeU lkbGtg"
        >
          Enter your city
        </label>
        <input
          aria-label="city"
          class="sc-dmqHEX jEDZgU"
          name="city"
          required=""
          type="text"
          value=""
        />
        <button
          class="sc-jTrPJq GpDmH"
          type="submit"
        >
          Reserve Now
        </button>
      </form>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
